hack/verify-contextual-logging.sh

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The KCP Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o nounset
+set -o pipefail
+# Don't set errexit because logcheck exits with non-zero when it finds errors to fix.
+
+SCRIPT_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+LOG_FILE="./${SCRIPT_ROOT}/hack/logging-results"
+
+errs=$(logcheck -check-contextual "${SCRIPT_ROOT}"/pkg/... 2>&1)
+
+
+# If the log file doesn't exist, make it
+if [[ ! -f "${LOG_FILE}" ]]; then
+    echo "${errs}" > "${LOG_FILE}"
+fi
+
+err_cnt=$(echo "${errs}" | wc -l | xargs)
+file_cnt=$(cat "${LOG_FILE}" | wc -l | xargs)
+
+if [[ ${err_cnt} -gt ${file_cnt} ]]; then
+    echo "Non-contextual log usage went up, not down."
+    diff "${LOG_FILE}" <(echo "${errs}")
+    exit 1
+fi
+
+echo "Contextual logging usage went up, updating log file"
```

Comment:
When we increase our contextual logging usage, do we want to force people to update this file and check it in? i.e. should CI fail if the file is not up to date?

Reply:
IMO, yes. I'll get it committed once I've got pkg/api added to the overall list